GVFS/GVFS/CommandLine/DiagnoseVerb.cs

```diff
@@ -50,9 +50,9 @@ protected override void Execute(GVFSEnlistment enlistment)
                 this.WriteMessage(string.Empty);
                 this.WriteMessage("gvfs version " + ProcessHelper.GetCurrentProcessVersion());
 
-                GitVersion gitVersion;
-                string error;
-                if (GitProcess.TryGetVersion(out gitVersion, out error))
+                GitVersion gitVersion = null;
+                string error = null;
+                if (!string.IsNullOrEmpty(enlistment.GitBinPath) && GitProcess.TryGetVersion(enlistment.GitBinPath, out gitVersion, out error))

```

Comment:
Just to double check - is enlistment.GitBinPath always guaranteed to be initialized properly for `DiagnoseVerb`?

Reply:
Yes - GVFSVerb.CreateEnlistment sets it. When Git is not installed, gvfs diagnose will early exit.